symbols/overlay29.yml

```diff
@@ -8500,6 +8500,44 @@ overlay29:
         Called when the in-dungeon "others" menu is open. Does not return until the menu is closed.
         
         return: Always 0
+    - name: GetAiUseItemProbability
+      address:
+        EU: 0x231EAC4
+        NA: 0x231DEB8
+        JP: 0x231F138
+      description: |-
+        Called to get the probability of an item being used or thrown by an AI on the current turn.
+
+        r0: Pointer to either the user if it is an item used by the AI or the target if it is an item thrown by the AI
+        r1: Pointer to item
+        r2: Unknown 2-bit bitvector. Bit 1 influences only the Warp Seed case, while Bit 0 influences various items.
+        
+        return: Integer in range [0, 100]
+      - name: IsAdjacentToEnemy
+      address:
+        EU: 0x231F358
+        NA: 0x231E74C
+        JP: 0x231F9CC
+      description: |-
+        Called to check if a hostile entity is present in any of the tiles adjacent to an entity.
+
+        r0: Pointer to entity
+        
+        return: True if yes, false if no
+      - name: GetRelationBetweenMonsters
+      address:
+        EU: 0x2302188
+        NA: 0x230157C
+        JP: 0x23027FC
+      description: |-
+        Called to check if two monsters are allies, enemies, or neither (one is a shopkeeper, decoy, etc.)
```

Comment:
This looks like a [similar function in Rescue Team](https://github.com/pret/pmd-red/blob/master/src/dungeon_ai_targeting.c#L463). There's more nuance to the return values than it may initially seem.

To be more precise, this function returns whether the first entity should _treat_ the second Pokémon as an ally, enemy, or ignore them. The complexity here is with decoy handling, because Pokémon prioritize targeting a decoy over other Pokémon. In addition, decoy handling is handled somewhat differently if a teammate or a wild Pokémon becomes a decoy. Wild decoys will be targeted by other wild Pokémon, while teammates won't be targeted by other teammates (thankfully). The decoy targeting is accomplished by returning different values from this function if a decoy is present, such that Pokémon target the decoy and ignore other Pokémon that they would normally target.

Aside from a few (well, quite a few) conditions at the beginning, most of the decisions are made by indexing into a 4-D array based on a number of conditions: the "decoy targeting mode" of the Pokémon, whether entity 1 is an enemy, whether entity 2 is an enemy, and whether entity 2 is a decoy. The array enumerates all possible combinations of these conditions, with the resulting targeting relation to use for each combination.

Reply:
"while teammates won't be targeted by other teammates (thankfully)."

Is this really the case? I definitely remember an enemy using Substitute or something similar on me and having my whole team target me lol

Reply:
If your team has AOE moves then they may hit friendly decoys accidentally. I believe they don't target you directly with single-target moves.